File: SP-UI/src/app/models/category.ts
import { MenuProduct, Product } from "./product"

export interface Category {
    id: number
    name: string
    image: string
    departmentId: number
    departmentName: string
}

export interface MenuCategory {
    id: number
    name: string
    products: MenuProduct[]
}

export interface PageCategory {
    id: number
    name: string
    image: string
}

export interface CategoryCreationDTO {
    name: string
    image: File
    departmentId: number
}

File: SP-API/SP-API/Entities/Category.cs
using System.ComponentModel.DataAnnotations;

namespace Mall_API.Entities
{
    public class Category
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        public string Image { get; set; }
        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "You must select a {0} from the list")]
        [Display(Name = "Department")]
        public int DepartmentId { get; set; }
        public Department Department { get; set; }
        public ICollection<Product> Products { get; set; }
    }
}
